### PR TITLE
Disambiguate `getActiveFiles()` methods in TransactionLogAccess

### PR DESCRIPTION
**Pull request by findinpath**
## Description
Code cosmetics PR for Delta Lake connector.

### CODE REVIEW

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java [hunk 1]
```diff
@@ -239,6 +238,10 @@ public Stream<AddFileEntry> getActiveFiles(ConnectorSession session, TableSnapsh
         return retrieveActiveFiles(session, tableSnapshot, metadataEntry, protocolEntry, TupleDomain.all(), alwaysTrue());
     }
 
+    /**
+     * Retrieves the active file entries matching the specified partition constraint
+     * along with the column statistics for the specified projected columns only.
```
Comment (findepi) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java:243]: javadoc should also say what `projectedColumns = Optional.empty` means.

(and per offline discussion, it seems that we don't need Optional at all, if Optional.empty means the same thing as Optiona.of(emptySet)

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java [hunk 2]
```diff
@@ -227,38 +227,26 @@ public MetadataEntry getMetadataEntry(ConnectorSession session, TableSnapshot ta
...
             MetadataEntry metadataEntry,
             ProtocolEntry protocolEntry,
             TupleDomain<DeltaLakeColumnHandle> partitionConstraint,
-            Optional<Set<DeltaLakeColumnHandle>> projectedColumns)
+            Set<DeltaLakeColumnHandle> projectedColumns)
     {
         Predicate<String> addStatsMinMaxColumnFilter = alwaysFalse();
-        if (projectedColumns.isPresent()) {
-            Set<String> baseColumnNames = projectedColumns.get().stream()
+        if (!projectedColumns.isEmpty()) {
```
Comment (findepi) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java:239]: the `if` condition is redundant.
you can always execute the block

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java [hunk 3]
```diff
@@ -1379,7 +1381,13 @@ public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle
...
         if (!newColumnMetadata.isNullable()) {
             boolean tableHasDataFiles;
-            try (Stream<AddFileEntry> addFileEntries = transactionLogAccess.getActiveFiles(session, getSnapshot(session, handle), handle.getMetadataEntry(), handle.getProtocolEntry())) {
+            try (Stream<AddFileEntry> addFileEntries = transactionLogAccess.getActiveFiles(
+                    session,
+                    getSnapshot(session, handle),
+                    handle.getMetadataEntry(),
+                    handle.getProtocolEntry(),
+                    TupleDomain.all(),
+                    alwaysFalse())) {
```
Comment (pajaks) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java:1390]: This was `alwaysTrue()` before. It this change intended? Maybe it should be in separate commit?

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/VacuumProcedure.java [hunk 4]
```diff
@@ -200,7 +202,12 @@ private void doVacuum(
             List<Long> recentVersions = transactionLogAccess.getPastTableVersions(fileSystem, transactionLogDir, threshold, tableSnapshot.getVersion());
             Set<String> retainedPaths;
             try (Stream<AddFileEntry> activeAddEntries = transactionLogAccess.getActiveFiles(
-                    session, tableSnapshot, handle.getMetadataEntry(), handle.getProtocolEntry())) {
+                    session,
+                    tableSnapshot,
+                    handle.getMetadataEntry(),
+                    handle.getProtocolEntry(),
+                    TupleDomain.all(),
+                    alwaysFalse())) {
```
Comment (pajaks) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/procedure/VacuumProcedure.java:210]: This was alwaysTrue() before. It this change intended? Maybe it should be in separate commit?

**Review by pajaks — APPROVED** [review]
